javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -36,81 +46,347 @@ object Long {
-    if (x == y) 0
-    else if (x < y) -1
-    else 1
+  @inline def min(a: scala.Long, b: scala.Long): scala.Long =
+    Math.min(a, b)
+
+  def numberOfLeadingZeros(l: scala.Long): Int =
+    Intrinsics.`llvm.ctlz.i8`(l.toByte, iszeroundef = false)
+
+  def numberOfTrailingZeros(l: scala.Long): Int =
+    Intrinsics.`llvm.cttz.i8`(l.toByte, iszeroundef = false)
+
+  @inline def parseLong(s: _String, radix: Int): scala.Long = {
+    val length   = s.length()
+    val negative = s.charAt(0) == '-'
+
+    if (s == null || radix > Character.MIN_RADIX ||
+        radix > Character.MAX_RADIX) throw new NumberFormatException(s)
+    if (length == 0) throw new NumberFormatException(s)
+    if (negative && length == 1) throw new NumberFormatException(s)
+
+    parse(s, 1, radix, negative)
   }
 
-  @inline def compareUnsigned(x: scala.Long, y: scala.Long): scala.Int = ???
+  @inline def parseLong(s: _String): scala.Long =
+    parseLong(s, 10)
 
-  def divideUnsigned(dividend: scala.Long, divisor: scala.Long): scala.Long =
-    divULong(dividend, divisor)
+  def parseUnsignedLong(s: _String): scala.Long =
+    parseUnsignedLong(s, 10)
+
+  @inline def parseUnsignedLong(s: _String, radix: Int): scala.Long =
+    undefined
 
+  @inline
   def remainderUnsigned(
       dividend: scala.Long, divisor: scala.Long): scala.Long =
     remULong(dividend, divisor)
 
-  def highestOneBit(i: scala.Long): scala.Long = ???
+  def reverse(l: scala.Long): scala.Long =
+    undefined
 
-  def lowestOneBit(i: scala.Long): scala.Long = ???
+  def reverseBytes(i: scala.Long): scala.Long =
+    undefined
 
-  def bitCount(i: scala.Long): scala.Int = ???
+  def rotateLeft(i: scala.Long, distance: scala.Int): scala.Long =
+    undefined
 
-  def reverseBytes(i: scala.Long): scala.Long = ???
+  def rotateRight(i: scala.Long, distance: scala.Int): scala.Long =
+    undefined
 
-  def rotateLeft(i: scala.Long, distance: scala.Int): scala.Long = ???
+  def signum(i: scala.Long): scala.Int =
+    if (i == 0) 0
+    else if (i < 0) -1
+    else 1
 
-  def rotateRight(i: scala.Long, distance: scala.Int): scala.Long = ???
+  @inline def sum(a: scala.Long, b: scala.Long): scala.Long =
+    a + b
 
-  def signum(i: scala.Long): Int = ???
+  def toBinaryString(l: scala.Long): _String = {
+    var count = 1
+
+    if (l < 0) {
+      count = 64
+    } else {
+      var k = l
+      while (k != 0) {
+        count += 1
+        k >>= 1
+      }
+    }
+
+    var j = l
+    val buffer = new Array[Char](count)
+    do {
+      count -= 1
+      buffer(count) = ((j & 1) + '0').toChar
+      j >>= 1
+    } while (count > 0)
+
+    new _String(0, buffer.length, buffer)
+  }
 
-  def numberOfLeadingZeros(l: scala.Long): Int = ???
+  def toHexString(l: scala.Long): _String = {
+    var count = 1
+
+    if (l < 0) {
+      count = 16
+    } else {
+      var k = l
+      while (k != 0) {
+        count += 1
+        k >>= 4
+      }
+    }
```

Comment:
Here it's a bit more complicated due to the ceil division by 4, but still possible:
```scala
var count = if (l == 0L) 1 else ((64 + 3) - numberOfLeadingZeros(l)) >> 2
```